### PR TITLE
ISPN-14008 query and annotations docs review

### PR DESCRIPTION
**Pull request by dvagnero**
Docs review of the query and indexing updates. Revision of indexing mode and index sharding (Infinispan 15 only)

needs backport to 14

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/code_examples/BookEmbedded.java
Comment (dvagnero) [inline documentation/src/main/asciidoc/topics/code_examples/BookEmbedded.java:1]: @fax4ever  Is it necessary to specify the structure for the @Embedded in this example? We don't have an example that illustrates the usage of nested or flattened

File: documentation/src/main/asciidoc/topics/protobuf/library.proto [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//needs review
```
Comment (dvagnero) [inline documentation/src/main/asciidoc/topics/protobuf/library.proto:1]: @fax4ever Can you please take a look and update this example?

**Comment by dvagnero** [comment]
@fax4ever I made another revision and made few adjustments. Please take a look at the changes and the two examples I commented on. Let me know what you think.

File: documentation/src/main/asciidoc/topics/code_examples/BookEmbedded.java
Comment (fax4ever) [reply documentation/src/main/asciidoc/topics/code_examples/BookEmbedded.java:1]: > @fax4ever Is it necessary to specify the structure for the @Embedded in this example?

It is not. Where not specified it means Structure.NESTED that is default for this annotation value.

> We don't have an example that illustrates the usage of nested or flattened

At the moment the Ickle query language does not allow to fully exploit the potential differences between the two options here. So I think it is OK.

File: documentation/src/main/asciidoc/topics/protobuf/library.proto (hunk 1 above)
Comment (fax4ever) [reply documentation/src/main/asciidoc/topics/protobuf/library.proto:1]: Yes we need to upgrade with the new annotations. I missed it last time :)

Comment (fax4ever) [reply documentation/src/main/asciidoc/topics/protobuf/library.proto:1]: With the new annotations it becomes:

``` proto
package book_sample;

/* @Indexed */
message Book {

    /* @Text(projectable = true) */
    optional string title = 1;

    /* @Text(projectable = true) */
    optional string description = 2;

    // no native Date type available in Protobuf
    optional int32 publicationYear = 3; 

    repeated Author authors = 4;
}

message Author {
    optional string name = 1;
    optional string surname = 2;
}
```

File: documentation/src/main/asciidoc/topics/proc_query_using_analyzers.adoc [hunk 2]
```diff
@@ -5,8 +5,8 @@ You specify analyzer definitions with the `@Text` annotation in your Java classe
 
 .Procedure
 
-. Include the `Analyze.YES` attribute to indicate that the property is analyzed.
-. Specify the analyzer definition with the `@Analyzer` annotation.
+. Include the `analyzer` attribute to indicate that the property is analyzed.
```
Comment (fax4ever) [inline documentation/src/main/asciidoc/topics/proc_query_using_analyzers.adoc:8]:
```suggestion
. Annotate the property with a `@Text` annotation to indicate that the property is analyzed.
```

File: documentation/src/main/asciidoc/topics/proc_query_using_analyzers.adoc [hunk 3]
```diff
@@ -5,8 +5,8 @@ You specify analyzer definitions with the `@Text` annotation in your Java classe
...
+. Specify the desired analyzer that you want to use for indexing and searching.
```
Comment (fax4ever) [inline documentation/src/main/asciidoc/topics/proc_query_using_analyzers.adoc:9]:
```suggestion
. Specify the desired analyzer that you want to use for indexing and searching using the `analyzer` attribute
```

**Review by fax4ever — COMMENTED** [review]
Here is my suggestions

**Comment by dvagnero** [comment]
@fax4ever awesome. Thank you for the examples and the suggestions. I updated the content and prepared backport

**Comment by fax4ever** [comment]
@dvagnero I think that we have some conflicts here

**Comment by dvagnero** [comment]
I'm sorry @fax4ever now it should be fixed

**Comment by fax4ever** [comment]
Merged thanks

**Pull request merged**